statement.k

```diff
@@ -16,17 +17,77 @@ rule if (false) S:Statement => .
 rule if (true) S:Statement else S1:Statement => #exeStmt(S)
 rule if (false) S:Statement else S1:Statement => #exeStmt(S1)
 
-rule #exeStmt({ Ss:Statements }) =>  Ss 
+rule #exeStmt({ Ss:Statements }) =>  Ss
+
+rule #exeStmt(S:IfStatement) => S
+
+rule #exeStmt(S:WhileStatement) => S
 
-rule #exeStmt(S:IfStatement) => S 
 
-rule _:Value ; => . 
+rule _:Value ; => .
 
 rule
     <k> #exeStmt(S:SimpleStatement;) => S; ...</k>
     <callStack> .List </callStack>
 
 rule
-   #localFunctionCall(FN:Id, ARGS:Values) => functionCall(FN; ARGS)      [structural]
-   
+    #localFunctionCall(FN:Id, ARGS:Values) => functionCall(FN; ARGS)      [structural]
+
+rule [While]:
+    while (E:Expression) S:Statement => if (E) { S #loop_mark while (E) S }      [structural]
+
+rule [Normal-Loop-Mark]:
+    <k> #exeStmt(#loop_mark) => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> false </exceptionMode>
+    </callState>
+
+rule [Exception-Skip]:
+    <k> S:Statement => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> true </exceptionMode>
+    </callState>
+
+rule [Exception-Continue-Loop-Mark]:
+    <k> #exeStmt(#loop_mark) => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> true => false </exceptionMode>
+        <exceptionCode> #continue# => .K </exceptionCode>
+    </callState>
+
+rule [Continue]:
+    <k> #exeStmt(S:Continue) => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> _:Bool => true </exceptionMode>
+        <exceptionCode> _ => #continue# </exceptionCode>
+    </callState>
+
+rule [Break]:
+    <k> #exeStmt(S:Break) => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> _:Bool => true </exceptionMode>
+        <exceptionCode> _ => #break# </exceptionCode>
+    </callState>
+
+rule [Exception-Break-Loop-Mark]:
+    <k> ( #exeStmt(#loop_mark) ~> S:WhileStatement .Statements) => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> true => false </exceptionMode>
+        <exceptionCode> #break# => .K </exceptionCode>
+    </callState>
+
+rule [Exception-Return-Loop-Mark]:
+    <k> ( #exeStmt(#loop_mark) ~> S:WhileStatement .Statements) => . ... </k>
+    <callState>
+        ...
+        <exceptionMode> true </exceptionMode>
+        <exceptionCode> #return# </exceptionCode>
```

Comment:
Why do you need a special case to handle the `return` statement? 
Shouldn't `return` be pretty much orthogonal/independent from loops? 
I mean, `return` should... return from a function, regardless of whether you are in a loop or not, I think;

Reply:
Yes, you are wright. I have removed the rule.